cfme/modeling/base.py

```diff
@@ -0,0 +1,143 @@
+import attr
+from collections import Callable
+
+from cached_property import cached_property
+
+from cfme.utils.appliance import NavigatableMixin
+
+
+class ApplianceCollections(object):
+    """Caches instances of collection objects for use by the collections accessor
+
+    The appliance object has a ``collections`` attribute. This attribute is an instance
+    of this class. It is initialized with an appliance object and locally stores a cache
+    of all known good collections.
+    """
+    _collection_classes = None
+
+    def __init__(self, appliance):
+        self._collection_cache = {}
+        self.appliance = appliance
+        if not self._collection_classes:
+            self.load_collections()
+        for collection, cls in self._collection_classes.items():
+            self._collection_cache[collection] = cls(self.appliance)
+
+    def load_collections(self):
+        """Loads the collection definitions from the entrypoints system"""
+        from pkg_resources import iter_entry_points
+        ApplianceCollections._collection_classes = {
+            ep.name: ep.resolve() for ep in iter_entry_points('manageiq.appliance_collections')
+        }
+
+    def __getattr__(self, name):
+        try:
+            return self._collection_cache[name]
+        except KeyError:
+            raise Exception('Collection [{}] not known to applinace'.format(name))
+
+
+class ObjectCollections(ApplianceCollections):
```

Comment:
i propose to have ` class ObjectCollections` as a base class taking the filter spec as a ctor arg


appliance collections should derive the filter spec from the  entrypoint (they just wouldn't be filters in that case)

Reply:
You can't, because filters are different for each collection that is instantiated, the filter isn't standard for the entire object. At least not necessarily